url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/MainFrame.java

```diff
@@ -0,0 +1,588 @@
+
+						} catch (IOException e1) {
+							// TODO Auto-generated catch block
+							e1.printStackTrace();
+						}
+						break;
+						
+						
+					case JOptionPane.NO_OPTION :
+						break;
+						
+					case JOptionPane.CANCEL_OPTION :
+						return;
+					}
+					
+					
+					JFileChooser fc = new JFileChooser();
+					fc.setFileFilter(new FileNameExtensionFilter("Emploi du Temps", "EdT"));
+					int res = fc.showOpenDialog(null);
+					File f = null;
+					if (res == JFileChooser.APPROVE_OPTION)	{
+						// On a le fichier.
+						f = fc.getSelectedFile();
+						// On sauvegarde sa localisation pour la prochaine ouverture du programme.
+						File dataStoreLocation = new File(System.getProperty("user.dir") + "/dataStoreLocation.loc");
+						BufferedWriter writer;
+						try {
+							writer = new BufferedWriter(new FileWriter(dataStoreLocation));
+							writer.write (f.getAbsolutePath()); // On écrit le chemin du DataStore
+							writer.close();
+						} catch (IOException e1) {
+							// TODO Auto-generated catch block
+							JOptionPane.showMessageDialog(null, "Impossible de sauvegarde la localisation du fichier de sauvegarde.", "Erreur de sauvegarde", JOptionPane.ERROR_MESSAGE);
+						}
+
+						// On essaie maintenant de charger le DataStore
+						try
+						{
+							FileInputStream inputFileStream = new FileInputStream(f.getAbsolutePath());
+							ObjectInputStream objectInputStream = new ObjectInputStream(inputFileStream);
+							dataStore = (DataStore)objectInputStream.readObject();
+							objectInputStream.close();
+							inputFileStream.close();
+						}
+						catch(ClassNotFoundException e1)
+						{
+							JOptionPane.showMessageDialog(null, "Erreur. Le fichier de sauvegarde semble corrumpu. Veuillez réessayer ou recréer une nouvelle base de données.", "Fichier de sauvegarde corrompu", JOptionPane.ERROR_MESSAGE);
+							new WelcomeFrame();
+						}
+						catch(IOException i)
+						{
+							JOptionPane.showMessageDialog(null, "Fichier de sauvegarde introuvable", "Le fichier de sauvegarde est introuvable ; merci de le localiser à nouveau.", JOptionPane.ERROR_MESSAGE);
+							new WelcomeFrame();
+						}
+
+						new MainFrame(dataStore);
+					}
+					setVisible(false);
+				}
+			});
+			
+			btnNew = new JButton(new ImageIcon(ImageIO.read(new File("img/icons/table_add.png"))));
+			btnNew.setText("Nouveau");
+			btnNew.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					int n = JOptionPane.showConfirmDialog(null, "Voulez sauvegarder la session en cours ?", "Attention", JOptionPane.YES_NO_CANCEL_OPTION);
+					
+					switch(n)	{
+					case JOptionPane.YES_OPTION :
+						BufferedReader br;
+						try {
+							br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Execution paths that do not contain closure statements were detected. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).